librad/src/net/peer.rs

```diff
@@ -33,29 +33,54 @@ pub use storage::Storage as PeerStorage;
 pub struct Config<Signer> {
     pub signer: Signer,
     pub protocol: protocol::Config,
-    pub storage_pools: PoolSizes,
+    pub storage: config::Storage,
 }
 
-#[derive(Clone, Copy)]
-pub struct PoolSizes {
-    /// Number of [`git::storage::Storage`] instances to pool for [`Peer`]
-    /// consumers.
+pub mod config {
```

Comment:
One (unrelated) general question I have is: when do we want to inline `mod`s into files vs creating a new file? The diffs when eventually moving to a file becomes kind of annoying to read so I think I fall into the preference of just going with the file structure. One exception I don't mind is the `pub mod error` pattern. What do yis think?

Reply:
I'd prefer to employ a discipline of doing shuffles in a separate commit (I can split this one if you want). It's just the nature of ML-style languages that one needs to cut the papyrus every so often, imho. Otherwise where do you stop? With enough code accumulated, you'll converge to Java-style one-type-per-file eventually, which is no longer navigable without IDE support.

Reply:
> I can split this one if you want

Nah, alles klar. Finding the right module split has always haunted me :sweat_smile: I can see pros on both sides of the argument :)